Lib/inspect.py

```diff
@@ -2835,23 +2835,32 @@ class BoundArguments:
         Dict of keyword arguments values.
     """
 
-    __slots__ = ('arguments', '_signature', '__weakref__')
+    __slots__ = ('arguments', '_signature', '_kwarg_names', '__weakref__')
 
-    def __init__(self, signature, arguments):
+    def __init__(self, signature, arguments, kwarg_names={}):
         self.arguments = arguments
         self._signature = signature
+        self._kwarg_names = kwarg_names
```

Comment:
```suggestion
    def __init__(self, signature, arguments, kwarg_names=None):
        self.arguments = arguments
        self._signature = signature
        self._kwarg_names = kwarg_names or {}
```

Should not use mutable instances as function defaults.

Reply:
Thanks, don't know what I was thinking about then..